src/routers/user.js

```diff
@@ -0,0 +1,192 @@
+const express = require("express");
+const User = require("../models/user");
+const auth = require("../middleware/auth");
+const auth_admin = require("../middleware/auth_admin");
+const { sendWelcomeEmail, sendCancelEmail } = require("../emails/account");
+const router = new express.Router();
+const multer = require("multer");
+const sharp = require("sharp");
+
+//REST APIs Route for Create single user
+router.post("/users", auth_admin, async (req, res) => {
+  try {
+    const user = new User(req.body);
+    sendWelcomeEmail(user.email, user.name);
+    await user.save();
+    res.status(201).send(user);
+  } catch (e) {
+    res.status(400).send(e);
+  }
+});
+
+//Doing User LogIn
+router.post("/users/login", async (req, res) => {
+  try {
+    const user = await User.findByCredentials(
+      req.body.email,
+      req.body.password
+    );
+    const token = await user.generateAuthToken();
+    res.send({ user, token });
+  } catch (e) {
+    res.status(400).send();
+    console.log(e);
+  }
+});
+
+//LogOut(by removing the item from given array)
+router.post("/users/logout/me", auth, async (req, res) => {
+  try {
+    console.log(req.user);
+    req.user.tokens = req.user.tokens.filter((token) => {
+      return token.token !== req.token;
+    });
+    await User.updateOne(
+      { _id: req.user._id },
+      {
+        $set: { isLogIn: false },
+      }
+    );
```

Comment:
why you created **isLogIn** field in your mongoose schema?
Think ,if you are login on 3 plus devices. if one device perform logout then ?????
This is not a better way. You have to expire your jwt token there.